thorlcr/activities/indexread/thindexreadslave.cpp

```diff
@@ -952,12 +952,10 @@ class CIndexCountSlaveActivity : public CIndexReadSlaveBase, public CThorDataLin
                     }
                 }
                 else
-                {
                     totalCount += (rowcount_t)klManager->getCount();
-                    if ((totalCount > choosenLimit))
-                        break;
-                }
                 callback.clearManager();
+                if ((totalCount > choosenLimit))
+                    break;
```

Comment:
NB: moving this check, fixes the crash by ensuring the manager is cleared before leaving, but also breaks out earlier in the hasFilter() case, which would have otherwise gone through all parts.